app/vmauth/main.go

```diff
@@ -198,7 +199,12 @@ func processRequest(w http.ResponseWriter, r *http.Request, ui *UserInfo) {
 				return
 			}
 			missingRouteRequests.Inc()
-			httpserver.Errorf(w, r, "missing route for %s", u.String())
+			dump, err := httputil.DumpRequest(r, false)
+			if err != nil {
+				httpserver.Errorf(w, r, "failed to dump HTTP request: %s", err)
+				return
+			}
+			httpserver.Errorf(w, r, "missing route for %q (%q)", u.String(), dump)
```

Comment:
Could it expose authorization header values and other sensitive headers information to the logs?

Reply:
Yes. We can make dumping optional, for debug purposes.

Reply:
Updated PR